androidComposePrototype/src/androidTest/java/com/genesys/cloud/messenger/uitest/test/ComposePrototypeUITest.kt

```diff
@@ -248,4 +309,116 @@ class ComposePrototypeUITest : BaseTests() {
         }
         bye()
     }
+
+    @Test
+    fun testUnknownAgent() {
+        apiHelper.disconnectAllConversations()
+        enterDeploymentInfo(testConfig.humanizeDisableDeploymentId)
+        connect()
+        sendMsg(helloText)
+        val conversationInfo = apiHelper.answerNewConversation()
+        if (conversationInfo == null) AssertionError("Unable to answer conversation.")
+        else {
+            Log.i(TAG, "Conversation started successfully.")
+            apiHelper.sendOutboundMessageFromAgentToUser(conversationInfo, outboundMessage)
+            verifyResponse(outboundMessage)
+            verifyResponse(humanizeDisabledText)
+            verifyResponse(humanText)
+            apiHelper.sendConnectOrDisconnect(conversationInfo, false, true)
+        }
+        bye()
+    }
+
+    @Test
+    fun testBotAgent() {
+        apiHelper.disconnectAllConversations()
+        enterDeploymentInfo(testConfig.botDeploymentId)
+        connect()
+        verifyResponse(botImageName)
+        verifyResponse(botEntity)
+        sleep(3000)
+        sendBotResponseAndCheckReply(yesText)
+        bye()
+    }
+
+    @Test
+    fun testDisconnectAgent_ReadOnly() {
+        apiHelper.disconnectAllConversations()
+        enterDeploymentInfo(testConfig.agentDisconnectDeploymentId)
+        connect()
+        sendMsg(helloText)
+        val conversationInfo = apiHelper.answerNewConversation()
+        if (conversationInfo == null) AssertionError("Unable to answer conversation.")
+        else {
+            Log.i(TAG, "Conversation started successfully.")
+            apiHelper.sendConnectOrDisconnect(conversationInfo, false, true)
+            // wait for agent to disconnect
+            apiHelper.waitForParticipantToConnectOrDisconnect(conversationInfo.id, "disconnected")
+            checkForReadOnly()
```

Comment:
There are 3 options for conversation disconnect configurations.
1. Do NOT send Event.ConversationDisconnect and do NOT transition State.ReadOnly
2. Send Event.ConversationDisconnect but do NOT transition to State.ReadOnly
3. Send Event.ConversationDisconnect AND transition to State.ReadOnly

Are there a way to test all of this scenarios including that `ConversationDisconnect` event is not send when scenario 1 is configured and IS sent for scenario 2 and 3, as well as verify that Transport does make state transition as expected?

Reply:
The new tests cover the 2 scenarios that send the ConversationDisconnect and make sure the history is updated. For scenario #1, we will update our Bot test to make sure the ConversationDisconnect event is not received since that deployment is not configured to send ConversationDisconnect event.